plugins/fbc-art-images/commands/setup.md

```diff
@@ -0,0 +1,209 @@
+---
+description: Generate CatalogSource, IDMS, and configure pull-secret for FBC testing
+argument-hint: <ocp-version> <operator-name>
+---
+
+## Name
+fbc-art-images:setup
+
+## Synopsis
+```
+/fbc-art-images:setup <ocp-version> <operator-name>
+```
+
+## Description
+The `fbc-art-images:setup` command automates the setup process for testing operators built in FBC (File-Based Catalog) by Konflux. It generates all necessary Kubernetes manifests and updates the cluster configuration to enable testing with art-image-share registry images.
+
+This command orchestrates two key operations:
+1. Creates a CatalogSource manifest pointing to the FBC image
+2. Generates an IDMS (ImageDigestMirrorSet) manifest for registry mirroring
+
+**Prerequisites**: The cluster pull-secret must be configured with art-image-share credentials before running this command. See `pull-secret.md` in the plugin directory for detailed instructions.
+
+The command is designed to streamline the workflow for OpenShift engineers testing operator releases.
+
+## Implementation
+
+### Step 1: Validate Inputs and Prerequisites
+- Verify `$1` (ocp-version) is provided and follows the format `X.Y` (e.g., "4.18")
+- Verify `$2` (operator-name) is provided and is a valid operator name
+- Confirm the user has cluster access by checking `oc whoami`
+- **Check pull-secret prerequisite**: Verify that the cluster's `pull-secret` contains a token that allows pulling images from `quay.io/redhat-user-workloads/ocp-art-tenant/art-images-share` 
+  - If it doesn't exist, show error:
+    ```
+    Error: Pull-secret doesn't have auth credentials for `quay.io/redhat-user-workloads/ocp-art-tenant/art-images-share`
+    Before using this plugin, you must configure the art-image-share pull-secret.
+    See pull-secret.md in the plugin directory for detailed instructions.
+    ```
+  - Exit pull-secret verification doesn't contain that token
+- Check if `oras` CLI is installed:
+  ```bash
+  which oras
+  ```
+- If `oras` not installed, guide the user to install it from https://oras.land/docs/installation
+- Check if `jq` is installed:
+  ```bash
+  which jq
+  ```
+
+### Step 2: Determine FBC Image URL
+- Construct the FBC image URL using the pattern:
+  ```
+  quay.io/redhat-user-workloads/ocp-art-tenant/art-fbc:ocp__<ocp-version>__<operator-name>
+  ```
+  Example: For OCP 4.20 and operator advanced-cluster-management, the URL would be:
+  ```
+  quay.io/redhat-user-workloads/ocp-art-tenant/art-fbc:ocp__4.20__advanced-cluster-management
+  ```
+
+### Step 3: Generate CatalogSource Manifest
```

Comment:
Is my understanding correct that the setup command exercises the same logic implemented by catalog-source and imds commands? if so, can DRY this by making that logic skills that then all three commands leverage.